### ISSUE TITLE
dot notation instead comma

### ISSUE THREAD

**Issue by raylight75** [pelmered/filament-money-field#60]
Can we use dot notation instead comma 1024,90 === 1024.90

**Comment by raylight75** [comment]
Yes sorry, it depends on the locale, my mistake. I think the default on Windows is a comma so is it possible to replace comma with dot without changing Windows locale.

**Comment by saspegas** [comment]
mask working good and showing correctly, but when i click save button it's saving wrong format. i fix it the problem like this:

```
MoneyInput::make('purchase_price')
  ->label('Alış Fiyatı')
  ->required()
  ->numeric(true)
  ->afterStateHydrated(function (Set $set, $state) {
      $set('purchase_price', number_format($state, 2, ',', '.'));
  })
  ->dehydrateStateUsing(function ($state) {
      return str_replace(['.', ','], ['', '.'], $state);
  }),
```

The problem is: It's showing and storing masked value, but it must be shown masked and recorded UNMASKED.

**Comment by pelmered** [comment]
> mask working good and showing correctly, but when i click save button it's saving wrong format. i fix it the problem like this:
> 
> ```
> MoneyInput::make('purchase_price')
>   ->label('Alış Fiyatı')
>   ->required()
>   ->numeric(true)
>   ->afterStateHydrated(function (Set $set, $state) {
>       $set('purchase_price', number_format($state, 2, ',', '.'));
>   })
>   ->dehydrateStateUsing(function ($state) {
>       return str_replace(['.', ','], ['', '.'], $state);
>   }),
> ```
> 
> The problem is: It's showing and storing masked value, but it must be shown masked and recorded UNMASKED.

OK, thank you for this report. I will investigate further. What locale and currency are you using?

**Comment by raylight75** [comment]
I have the same problem as saspegas said, and I solved it with number_format function too

**Comment by pelmered** [comment]
I'm working on a fix for this now.

**Comment by pelmered** [comment]
As there is a decent work around for this, I will work on this for the coming 2.0 version which will include a partial rewrite of this logic.

**Comment by sneycampos** [comment]
> The problem is: It's showing and storing masked value, but it must be shown masked and recorded UNMASKED.

I don't know what is the solution @pelmered is working but may be a good choice to have a method/flag/anything to "unMaskOnSave" or something else

**Comment by saspegas** [comment]
> > mask working good and showing correctly, but when i click save button it's saving wrong format. i fix it the problem like this:
> > ```
> > MoneyInput::make('purchase_price')
> >   ->label('Alış Fiyatı')
> >   ->required()
> >   ->numeric(true)
> >   ->afterStateHydrated(function (Set $set, $state) {
> >       $set('purchase_price', number_format($state, 2, ',', '.'));
> >   })
> >   ->dehydrateStateUsing(function ($state) {
> >       return str_replace(['.', ','], ['', '.'], $state);
> >   }),
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The problem is: It's showing and storing masked value, but it must be shown masked and recorded UNMASKED.
> 
> OK, thank you for this report. I will investigate further. What locale and currency are you using?

Sorry for the very late response, I did not see it as I did not receive any notification. tr-TR and TRY (from Turkey).
Turkey uses (.) as thousands separator and (,) as fractions separator unlike en-US.

My advice, the mask works properly, but it does it with the en-TR local setting while saving. But php and database are en-US. My advice is to use two variables. One is maskedValue and the other is raw value. maskedValue value should only be shown for the user to see. The real value should not be changed (it should use the raw values). In the current structure masked data is also used for saving.

**Comment by pelmered** [comment]
In version 2 of this package the internal state will be a money value object that contains all the information needed to transform the amount for both display and saving in the database. It will also utilise a custom model cast for this so it should be a lot more robust than the current solution.

**Comment by holmesadam** [comment]
When do you anticipate v2 will be released?

**Comment by pelmered** [comment]
@holmesadam It's getting closer now. The plan is to get a beta version out by the end of the week.
